### DISCUSSION TITLE
MessageBox actions

### DISCUSSION THREAD

**Discussion by Amadeus71r** [HASwitchPlate/openHASP#815] [Q&A] [answered]
The description of the msgbox object shows an example with two buttons "Apply" and "Close".

![image](https://github.com/user-attachments/assets/abca841e-2f27-4374-b26d-9c010eab2b7c)

I have an action button to call the msgbox object:
`{
"page":2,"id": 3,"x":50,"y":280,"w": 380,"h": 50,"obj": "btn","text": "Restart panel?","text_font":24,"action":{"up":{"page":0,"id":43,"obj":"msgbox","text":"Confirm?","options":["Apply", "Close"]}}
}`

Console response:
`p0b43 => {"event":"up","val":0,"text":"Yes"}`

How do I use the "Apply" button click event to reboot the device locally?

**Comment by fvanroie** [comment, marked as answer]
> Action commands are supported only by binary or visual type of objects and they are performed locally on the plate.

The `msgbox` object is a selector and doesn't support local `action` commands. You can redesign the popup as an `obj` as parent with a `label` and two `btn`s inside. Show/hide that object by using the buttons and local commands. Use the other button to reboot the device.

**Reply by Amadeus71r** [reply]
Thank's. I will try it. 

But I don't understand why to give such an example in the documentation...

Post-processing of events on the HA side, for example?

**Reply by fvanroie** [reply]
You can do what you want with HA automation too, but you asked specifically about **on the plate** only actions...
These are two different flows, the example is just that: an example. How you implement it depends on your use-case..

**Reply by Amadeus71r** [reply]
I understand. Thanks.
